src/python/dist.py

```diff
@@ -173,16 +175,32 @@ def distribute_discrete(sizes, groups, pow=1.0):
 
     off = 0
     curweight = 0.0
-    proc = 0
+
+    all_breaks = None
+    if breaks is not None:
+        # Check that the problem makes sense
+        all_breaks = np.unique(breaks)
```

Comment:
I think it is still useful to prune accidental duplicates in the input list.  We could construct a set from the breaks, which would also remove duplicates, but since we need a ndarray eventually anyway, I think this is fine...